### DISCUSSION TITLE
Replacement for add_object

### DISCUSSION THREAD

**Discussion by varunJainV** [FreeOpcUa/opcua-asyncio#1607] [Q&A] [answered]
Its taking time to add channel or object Is there any alternative to do the following:
![image](https://github.com/FreeOpcUa/opcua-asyncio/assets/160575473/60f481f3-cfa3-4ade-a6f5-5505412faa5c)

Code:
await extchannels.add_object(ua.NodeId(NamespaceIndex=ns),'2:ExternalChannel_%s' % num,external_channels)

Here I am using async await to add_object.
There are nearly 12 objects to add and it can be increased upto very high number of channels(objects), On an average its taking 1.5sec for each channel(object) to add just like the following screenshot.
![image](https://github.com/FreeOpcUa/opcua-asyncio/assets/160575473/1e953e33-57d0-4551-84a8-02d746f3f2f6)

**Comment by schroeder-** [comment, marked as answer]
You can use async, to run multiple corotines parallel like this:
```
async def create_object(extchannels, num):
     myobj1 = await extchannels.add_object(ua.NodeId(NamespaceIndex=ns),'2:ExternalChannel_%s' % num,external_channels)
     ....

coros = [create_object(extchannels, num) for num in range(1, channeldata + 1, 1)]
await asyncio.gather(*coros)
```

Othwise you can talk directly to the lower layer and make a modified version of https://github.com/FreeOpcUa/opcua-asyncio/blob/094860570227c28b23818e972a05432caf847b00/asyncua/common/manage_nodes.py#L207 and handle multiple add object calls at once.
